subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -4324,6 +4336,7 @@ static int generate_notify_message(struct lwm2m_ctx *ctx,
 	msg->token = obs->token;
 	msg->tkl = obs->tkl;
 	msg->reply_cb = notify_message_reply_cb;
+	msg->message_timeout_cb = notify_message_timeout_cb;
```

Comment:
How about using `ctx->notify_timeout_cb` directly here instead of adding an extra wrapper function? The current logic in lwm2m_engine would already verify if it's NULL or not.

Reply:
I thought about that. The problem is that struct lwm2m_message which is the argument to msg->message_timeout_cb is not exposed in lwm2m.h

Reply:
Ah, indeed, I have missed that. Ok then, let's keep it as is.